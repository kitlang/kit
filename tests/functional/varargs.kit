include "stdarg.h";

function f(num: Int, b...) {
    printf("args: %i\n", num);
    for i in 0 ... num {
        var x: Int = b;
        printf("%i\n", x);
    }
}

function g(num: Int, args...) {
    vf(num, (args...));
}

function vf(num: Int, args: va_list) {
    printf("args: %i\n", num);
}

function main() {
    f(3, 1, 2, 3);
    f(5, 4, 5, 6, 7, 8);
    g(2, 1, 2);
}
