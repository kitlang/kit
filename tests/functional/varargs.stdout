args: 3
1
2
3
args: 5
4
5
6
7
8
args: 2
